inbm/integration-reloaded/scripts/create_signature.py

```diff
@@ -18,12 +18,12 @@
     if num_params == 4:
         password = sys.argv[3].encode('utf-8')
 
-with open(package_name) as package:
-    checksum = hashlib.sha384(package.read().encode('utf-8')).hexdigest()
+with open(package_name, 'rb') as package:
+    checksum = hashlib.sha384(package.read()).hexdigest()
 
-with open(file_name) as f:
-    priv_key = load_pem_private_key(f.read().encode('utf-8'), password=password, backend=default_backend())
+with open(file_name, 'rb') as f:
+    priv_key = load_pem_private_key(f.read(), password=password, backend=default_backend())
```

Comment:
**Suggestion:** Avoid reading the entire private key file into memory, especially if it's large. Use serialization best practices for loading the private key. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
with open(file_name, 'rb') as f:
    priv_key = load_pem_private_key(f.read(2048), password=password, backend=default_backend())
```

Reply:
private key is not large; in either case, you always need the whole private key in memory